services/logs_collector_service.py

```
@@ -29,12 +31,137 @@ def __init__(self, api_url: Optional[str] = None, api_token: Optional[str] = Non
+                        # Add Host header for proper routing
+                        headers_with_host = headers.copy()
+                        headers_with_host["Host"] = hostname
+
+                        try:
+                            logger.info(f"Retrying with cached IP: {cached_ip}")
+                            response = requests.post(
+                                ip_endpoint,
+                                json=payload,
+                                headers=headers_with_host,
+                                timeout=5 + (attempt * 2),
+                                verify=False  # Skip SSL verification when using IP directly
```

Ruff (0.13.3)

Probable use of `requests` call with `verify=False` disabling SSL certificate checks

(S501)